sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -635,13 +635,19 @@ def append_taskrun_params(task_name_append: str):
                         if artifact_name == result['name'] and src != dst:
                             add_copy_results_artifacts_step = True
                             script += (
-                                    'ARTIFACT_SIZE=`wc -c %s | awk \'{print $1}\'`\n' % src +
+                                    'if [ -d ' + src + ' ]; then\n' + 
+                                    '  tar -czvf ' + src + '.tar.gz ' + src + '\n' +
+                                    '  SUFFIX=".tar.gz"\n' +
+                                    'fi\n' +
+                                    'ARTIFACT_SIZE=`wc -c %s${SUFFIX} | awk \'{print $1}\'`\n' % src +
                                     'TOTAL_SIZE=$( expr $TOTAL_SIZE + $ARTIFACT_SIZE)\n' +
                                     'touch ' + dst + '\n' +  # create an empty file by default.
                                     'if [[ $TOTAL_SIZE -lt 3072 ]]; then\n' +
-                                    '  if ! awk "/[^[:print:]]/{f=1} END{exit !f}" %s; then\n' % src +
+                                    '  if [ -d ' + src + ' ]; then\n' + 
+                                    '    tar -tzf ' + src + ' > ' + dst + '\n' +
```

Comment:
It looks like your generated tests doesn't match with the one generated by the test build. Also, I don't think Tekton can pass a directory as the Tekton results, are you sure this command will work in this case?

Reply:
Should I do anything except running ```make unit_test GENERATE_GOLDEN_YAML=True``` to generate the tests? I do not get why the code differs from what I wrote in _data_passing_rewriter.py file

Also, concerning passing directories, the fact that Tekton cannot pass a directory as the Tekton results is precisely the reason why I added the check: writing
```
'  if [ -d ' + src + ' ]; then\n' + 
'    tar -tzf ' + src + '.tar.gz > ' + dst + '\n' +
'  elif ! awk "/[^[:print:]]/{f=1} END{exit !f}" %s; then\n' % src +
'    cp ' + src + ' ' + dst + '\n' +
```
I make sure that, if the result is a directory, it is not copied in ```dst```, but instead only the content  of the compress archive is written in the file (I could have written ```'ls ' + src + ' > ' + dst '\n'``` probably).

You were right, though, there was a mistake that now I fixed